cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -488,8 +490,9 @@ func CreateControllerContext(s *config.CompletedConfig, rootClientBuilder, clien
 	versionedClient := rootClientBuilder.ClientOrDie("shared-informers")
 	sharedInformers := informers.NewSharedInformerFactory(versionedClient, ResyncPeriod(s)())
 
+	stopOnError := func(error) bool { return true }
```

Comment:
does this stop all informers used by anything in kube-controller-manager if there's a single list or watch error? what restarts them?

Reply:
No, it only stops metadata informers on a list or watch error. The typed informer factory continues to start informers via `Run` rather than `RunWithStopOptions` and thus these informers never get shut down.

Reply:
It does feel a little weird and non-obvious to have typed informers completely ignore the stopOnError, but I'm not sure there's a better work around given the controller-manager informer factory is awrapper around both typed informer factory and metadata informer factory.

Reply:
Let me know if you have any suggestions